pinot-core/src/main/java/org/apache/pinot/core/util/ListenerConfigUtil.java

```diff
@@ -112,6 +112,12 @@ private ListenerConfigUtil() {
 
     listeners.addAll(buildListenerConfigs(brokerConf, "pinot.broker.client", tlsDefaults));
 
+    // support legacy behavior < 0.7.0
+    if (listeners.isEmpty()) {
```

Comment:
You can build the ListenerConfigs from the config first, then append the HTTP config if the ListenerConfigs is empty?

Reply:
```suggestion
  public static List<ListenerConfig> buildBrokerConfigs(PinotConfiguration brokerConf) {
    TlsConfig tlsDefaults = TlsUtils.extractTlsConfig(brokerConf, CommonConstants.Broker.BROKER_TLS_PREFIX);
    List<ListenerConfig> listeners = buildListenerConfigs(brokerConf, "pinot.broker.client", tlsDefaults);
    if (listeners.isEmpty()) {
      int queryPort = brokerConf.getProperty(Helix.KEY_OF_BROKER_QUERY_PORT, Helix.DEFAULT_BROKER_QUERY_PORT);
      listeners = Collections.singletonList(
          new ListenerConfig(CommonConstants.HTTP_PROTOCOL, DEFAULT_HOST, queryPort, CommonConstants.HTTP_PROTOCOL,
              new TlsConfig()));
    }
    return listeners;
  }
```

Reply:
not sure I follow. We first attempt to create listeners from listener specs and an optionally set "port" variable. If neither exists, we fall back to a default spec.

Reply:
if someone defines "pinot.broker.client.queryPort", they expect to get an HTTP listener **even though** they may define a secondary https ingress via listeners